onnx/defs/nn/defs.cc

```diff
@@ -2832,4 +2832,470 @@ ONNX_OPERATOR_SET_SCHEMA(
             }));
+
+static const char* RMSNormalization_ver23_doc = R"DOC(
+      This is RMS normalization defined in ONNX as function as described in the paper https://arxiv.org/pdf/1910.07467.
+      The overall computation can be split into two stages. The first stage is standardization, which makes the
+      normalized elements have zero mean and unit variances. The root mean squared norm is taken over the last D dimensions,
+      where D is the dimension of normalized_shape. For example, if normalized_shape is (3, 5) (a 2-dimensional shape),
+      the rms norm is computed over the last 2 dimensions of the input. The computation required by standardization can be
+      described by the following equations.
+      ```
+      XSquared = Mul(X, X)
+      XSquaredMean = ReduceMean<axes=normalized_axes>(XSquared)
+      RMS = Sqrt(XSquaredMean)
+      RMSEps = Add(RMS, epsilon)
+      SqrtRMS = Sqrt(RMSEps)
+      Normalized = Div(X, SqrtRMS)
+      ```
+      where `normalized_axes` is `[axis, ..., rank of X - 1]`. The variables `RMS` stand for root mean square,
+      The second stage then scales and shifts the outcome of the first stage using:
+      ```
+      Y= Mul(Normalized, Scale)
+      ```
+      Let `d[i]` indicate the i-th dimension of `X`.
+      If `X`'s shape is `[d[0], ..., d[axis-1], d[axis], ..., d[rank-1]]`,
+      the shape of `RMS` is `[d[0], ..., d[axis-1], 1, ..., 1]`.
+      `Y` and `X` have the same shape. This operator supports unidirectional broadcasting
+      (tensors `Scale` and `B` should be unidirectional broadcastable to tensor `X`);
+      for more details please check [the doc](Broadcasting.md).
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    RMSNormalization,
+    23,
+    OpSchema()
+        .SetDoc(RMSNormalization_ver23_doc)
+        .Attr(
+            "axis",
+            "The first normalization dimension: normalization will be performed along dimensions axis : rank(inputs).",
+            AttributeProto::INT, static_cast<int64_t>(-1))
+        .Attr(
+            "epsilon",
+            "The epsilon value to use to avoid division by zero.",
+            AttributeProto::FLOAT, 1e-5f)
+        .Attr(
+            "stash_type",
+            "Type of Mean and InvStdDev. This also specifies stage one's computation precision.",
+            AttributeProto::INT,
+            static_cast<int64_t>(ONNX_NAMESPACE::TensorProto_DataType_FLOAT))
+        .AllowUncheckedAttributes()
+        .Input(0,
+               "X",
+               "The output of the layer for which the skip connection is being created. "
+               "In general, the shape is (N, C, D1, D2, ... , Dn) for n-dimensional data, where "
+               "D1 to Dn are the spatial dimension sizes and N is the batch size, C is the number of channels. "
+               "The root mean squared norm is taken over the last D dimensions, D is determined by the axis attribute.",
+               "T")
+        .Input(1,
+               "scale",
```

Comment:
Clarify the shape of scale. Is it required to be the normalized-shape, or can it be something that can be broadcast to the normalized-shape? Eg: is it allowed to be a scalar?